src/Libraries/Microsoft.Extensions.Http.Resilience/Resilience/Internal/StrategyKeyOptions.cs

```diff
@@ -1,19 +1,15 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
 using System.Net.Http;
 
 namespace Microsoft.Extensions.Http.Resilience.Internal;
 
 /// <summary>
 /// The provider that returns the strategy key from the request message.
 /// </summary>
-internal interface IStrategyKeyProvider
+internal sealed class StrategyKeyOptions
```

Comment:
Is this interface used in DI? If not, you could just be using Func<HttpRequestMessage, string> directly

Reply:
Yes, it comes from DI. This options class is just a holder for that delegate.